### PR TITLE
Added support of object initializers

### PR DESCRIPTION
**Pull request by metoule**
This PR adds support of object initializers, so that it's possible to write `new MyClass { MyProp = 5 }`.

### CODE REVIEW

File: test/DynamicExpresso.UnitTest/ConstructorTest.cs [hunk 1]
```diff
@@ -50,5 +50,73 @@ public void Unknown_New_Type_Is_Not_Supported()
...
+				StrProp = strValue;
+			}
+
+			public override bool Equals(object obj)
+            {
+                return Equals(obj as MyClass);
+            }
+			public bool Equals(MyClass p)
+			{
+				if (p is null) return false;
```
Comment (davideicardi) [inline test/DynamicExpresso.UnitTest/ConstructorTest.cs:116]: It seems that there is some code alignment (tab/space) issue ?

File: test/DynamicExpresso.UnitTest/ConstructorTest.cs [hunk 2]
```diff
@@ -50,5 +50,73 @@ public void Unknown_New_Type_Is_Not_Supported()
...
+			Assert.AreEqual(new MyClass() { StrProp = "test", IntField = 5 }, target.Eval("new MyClass() { StrProp = \"test\", IntField = 5 }"));
+		}
+
+		[Test]
+		public void Object_initializer_syntax_error()
+		{
+			var target = new Interpreter();
+			target.Reference(typeof(MyClass));
+
+			Assert.Throws<ParseException>(() => target.Parse("new MyClass() { StrProp }"));
```
Comment (davideicardi) [inline test/DynamicExpresso.UnitTest/ConstructorTest.cs:88]: Can you add some more tests for invalid cases, maybe like:
```
Assert.Throws<ParseException>(() => target.Parse("new MyClass() { StrProp ")); // no close bracket
Assert.Throws<ParseException>(() => target.Parse("new MyClass() StrProp }")); // no open bracket
Assert.Throws<ParseException>(() => target.Parse("new MyClass() {{IntField = 5}}")); // multiple bracket
Assert.Throws<ParseException>(() => target.Parse("new MyClass() {5}")); // no field name
```

**Review by davideicardi — CHANGES_REQUESTED** [review]
Great work! (thanks again ;-) )

I just added some very small comments.
If for you is fine, my plan will be to merge this PR and publish a new release that will contain several improvements thank to your work. Then in the next release we will add lambda syntax from the other PR. What do you think?

**Comment by metoule** [comment]
I fixed the formatting issues, and added the tests you suggested.
I like your plan to release the smaller changes before the lambda syntax PR, that way there's a fallback version in case something goes terribly wrong :)

**Review by davideicardi — APPROVED** [review]
Great!

**Pull request merged**